### DISCUSSION TITLE
rAudio 240219 - A total mess for me

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1605] [Fixed Issues]
I updated 240219 over 240212 avoiding the withdrawn 240218.
Effects:
1. invalid card number 'scontents' 
2. Setting "No Volume" => Output section in player configuration vanished.
3. No sound at all

I'm just trying to go back to 240212 on top of rAudio-64bit-20240108 image

**Comment by klassikmann** [comment]
Next try:
1. Use rAudio-64bit-20240108 + latest backup => everything o.k.
2. Update to 240219 => as described above

**Comment by cmh714** [comment]
I got the scontents message too....the Player page now is missing my DAC and everything in that section...

**Reply by ceejayemm** [reply]
This is an ongoing problem see (amongst others) #1594, #1600, #1602 and #1603. Some are reporting 20240219 fixes the problems but this isn't the case for me, although it is fractionally better with this release.

Chris

**Reply by klassikmann** [reply]
I'm trying step-by-step to approacg my newest working configuration.
Urgent question: How to update to an older release, e.g. update to 240212 instead to 240219?

**Reply by ceejayemm** [reply]
My recent update to #1594 might shed some light, if not a solution yet.

Chris

**Comment by cmh714** [comment]
I always try to have a reasonably current backup image for exactly these purposes!

**Comment by klassikmann** [comment]
@cmh714 
> I always try to have a reasonably current backup image for exactly these purposes!

Sure, I have enough backups (generated by the System-Menu).  But to restore not only such a **backup**, it would be helpful to restore/overload the existing **release** by an earlier one. My idea was to bibisect the release responsible for the regression. To make the situation (slighty) worse: I'm using the rAudio hardware for additional purposes also, e.g. one of two AdGuardHome instances in my infrastructure is co-located with rAudio.  But the full system is up and running again since two hours and a full backup of the 32GB SDcard is available.

**Reply by cmh714** [reply]
Understood on what you were attempting! And I make actual drive images in addition to the System generated backup. Make recovery a bit easier.

**Comment by rern** [comment]
Guys,

Please post your output of:
```sh
echo "\
defaults.pcm.card 0
defaults.ctl.card 0
" > /etc/asound.conf

aplay -l
```

**Comment by klassikmann** [comment]
@rern 
Before update to rAudio 20240219:
```
rAudio:~  #  cat /etc/asound.conf
defaults.pcm.card 0
defaults.ctl.card 0

rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_digi], device 0: HiFiBerry Digi+ Pro HiFi wm8804-spdif-0 [HiFiBerry Digi+ Pro HiFi wm8804-spdif-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

After update to rAudio 20240219:
```
Invalid card number 'scontents'
rAudio:~  #  cat /etc/asound.conf
defaults.pcm.card
defaults.ctl.card

rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
ALSA lib conf.c:1245:(parse_value) card is not a string
ALSA lib conf.c:2012:(_snd_config_load_with_include) _toplevel_:2:18:Invalid argument
ALSA lib conf.c:4120:(config_file_open) /etc/asound.conf may be old or corrupted: consider to remove or fix it
ALSA lib conf.c:4042:(snd_config_hooks_call) function snd_config_hook_load returned error: Invalid argument
ALSA lib conf.c:4649:(snd_config_update_r) hooks failed, removing configuration
aplay: device_list:288: control open (0): Invalid argument
```

After restore of /etc/asound.conf:
```
rAudio:~  #  echo "\
defaults.pcm.card 0
defaults.ctl.card 0
" > /etc/asound.conf
rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_digi], device 0: HiFiBerry Digi+ Pro HiFi wm8804-spdif-0 [HiFiBerry Digi+ Pro HiFi wm8804-spdif-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

After reboot:
```
rAudio:~  #  cat /etc/asound.conf
defaults.pcm.card
defaults.ctl.card

rAudio:~  #  aplay -l
**** List of PLAYBACK Hardware Devices ****
ALSA lib conf.c:1245:(parse_value) card is not a string
ALSA lib conf.c:2012:(_snd_config_load_with_include) _toplevel_:2:18:Invalid argument
ALSA lib conf.c:4120:(config_file_open) /etc/asound.conf may be old or corrupted: consider to remove or fix it
ALSA lib conf.c:4042:(snd_config_hooks_call) function snd_config_hook_load returned error: Invalid argument
ALSA lib conf.c:4649:(snd_config_update_r) hooks failed, removing configuration
aplay: device_list:288: control open (0): Invalid argument
```
Player settings: Device empty

**Comment by rern** [comment]
@klassikmann, @ceejayemm,

Please try the UPDATE.

**Comment by klassikmann** [comment]
@rern @ceejayemm 
First impression: Looks good.

**Comment by cmh714** [comment]
UPDATE sort of not working well here.....now my Device just says USB Audio not the actual device name. Unsure how big an issues tat may be....it plays.

Also it fails to mount 2 of my NFS drives. Going back to my backup image until this gets straightened out.

But still GREATLY appreciate @rerns efforts!!!!!!

**Comment by rern** [comment]
Please try the UPDATE again.

**Comment by klassikmann** [comment]
Still looking good.

**Reply by cmh714** [reply]
Not here....moving backwards :)

![image](https://github.com/rern/rAudio/assets/8241076/bcb66e3d-5983-4ae0-b922-26041a2b107f)

**Reply by cmh714** [reply]
after a reboot the DAC seems to have showed up properly....I think

**Comment by cmh714** [comment]
Also the UI shows 2 of my drives not mounted....but they ARE mounted and can play music...
![image](https://github.com/rern/rAudio/assets/8241076/8eab959c-cffd-4491-bf98-1a76dfba3abb)

**Reply by rern** [reply]
Please post the output of: `cat /etc/fstab`

**Reply by cmh714** [reply]
I would say we are not parsing spaces or backslashes properly at this time from what I see....as always Sir, TY! And again, the music plays fine so its just a cosmetic/UI thingy.

rAudio:~  #  cat /etc/fstab
PARTUUID=d062bfbc-01                            /boot                            vfat  defaults,noatime                  0  0
PARTUUID=d062bfbc-02                            /                                ext4  defaults,noatime                  0  0
192.168.2.131:/nfs/Public/Shared\040Music/Live  /mnt/MPD/NAS/Dead\040&\040Jerry  nfs   defaults,auto,bg,soft,timeo=5,ro  0  0
192.168.2.144:/nfs/Public/Shared\040Music       /mnt/MPD/NAS/Other\040Music      nfs   defaults,auto,bg,soft,timeo=5,ro  0  0
192.168.2.144:/nfs/MyPassport/Music             /mnt/MPD/NAS/Passport            nfs   defaults,auto,bg,soft,timeo=5,ro  0  0

**Reply by cmh714** [reply]
Update to 0312 seems to have fixed my minor drive issue.
TY @rern

**Comment by klassikmann** [comment]
rAudio 240226: Finally fixed.
